[#3763] Add some points regarding review process

dispera opened the pull request:
Scope
=====

Improve the review section.


Why we got into this (5 whys)
=============================

* I was not applying all the review instructions to Trac tickets.
* I mistakenly though most of the styleguide review process applied
only to Github, and was treating Trac issues review differently.
* I though this review applied more to code and for other tasks it was
not really very important.


Changes
=======

Made some additions to the Styleguide review page.


Reviewers
========

reviewers: @hcs0 @brunogola @dumol

brunogola commented on content/review.rst:
```diff
@@ -5,6 +5,9 @@ Review
 who requests a review and a person who checks the review.
+Please note that the same review concepts/procedures apply to all platforms
+(Github, Trac, email, etc) and not only to code changes, but to any task
+you are working on.
```
I think the last bit is redundant. What do you think about:
Please note that the process discussed here applies to all tickets and platforms (GitHub, Trac, email, etc), not only to code review.

dispera replied:
I trimmed it a bit and re-wrote it to look better now :)

brunogola commented on content/review.rst:
```diff
@@ -93,6 +96,10 @@ For the person requesting a review
   reviewer will check for sure.
 
+* Before submitting a ticket for review, check that you have documented your
+  work accordingly, like in a wiki page, or if you pushed a PR, list
```
I think we are trying to avoid using the wiki much, so maybe we should suggest using the styleguide or the documentation when documenting stuff?

dispera replied:
I see, I changed it then, thanks 👍

adiroiban replied:
As a sysadmin, you will still need to touch the private wiki... for ip addresses, service names ...etc

If possible, I would like to have the styleguide in such a way so that it will describe the "philosophy"  of the review process. rather than being a step by step (for dummies) guide for how to do a review using GitHub.

-------

Also, as part of the whole styleguide, we should include information and best practices based on our remote / async work ... and to help people understand what is specific to remote work, why we do one thing and not another.

hcs0 commented:
No other item to add other than those comments..

changes-approved

adiroiban commented:
I have removed myself from the list of reviewers.

@dispera don't forget to explicitly request the review for this PR

@brunogola don't forget to also add the explicit "changes-approved" comment ... as we are not yet using the GitHub review comments... due to missing API

@hcs0 there is no point in approving a review, if the review was not yet requested :)

It looks like this whole GitHub review workflow is badly documented ... or for some reason people are not able to understand how it works and why we have it in the first place

adiroiban commented:
so...based on the above message I think that this PR needs changes, as at least for Diego, Bruno and Hannah, the review process is not clarified

hcs0 commented:
@adiroiban Ah it needs the needs-review marker.!

dumol commented:
True, but Diego shoukd use that marker here, when the pull request is ready for review… Thus, as Adi requested above, this **needs-changes**. Don't worry, you'll pick up this awkward GitHub-Trac sync soon enough. ;-]

dispera commented:
The truth is, as I was not 100% convinced yet neither on how this is at the moment and wanted to wait and try to improve it further (maybe inspiration comes if I just leave it a day or two), I may wait a bit for requesting the review :)

Of course if anyone happens stops by meanwhile, comments are accepted anyway hahah,

dispera commented:
I rewrote the part I added to the intro. Please review the changes and see if you have something to correct or add to clarify the review process.

To sum up, this started when I submitted a trac issue for review without documenting my work anywhere else than the trac issue itself. I though that could be done in a separate issue in a later time. So the simple additions aim to help me and others on how to approach trac reviews.

needs-review

hcs0 commented:
Under **Overview of the Github and Trac integration**

Please add this sentence after:

_Integration is mainly between GitHub Pull Requests and Trac tickets, following the workflow described in review._

The sentence is:

_Please do not use or rely on the new GitHub PR review features as they have no API and is not integrated in Trac._

Based on this comment:
https://github.com/chevah/agent-1.5/pull/56#issuecomment-268601252

And this recent PR which had Adi R as a reviewer on the Github web UI but his review is not required to pass pqm https://github.com/chevah/server/pull/1108


needs_changes

dispera commented:
Great addition, Hannah :)
Added GitHub PR features note as per your comment,

That is exactly what was showing me I approved the changes on your PR early this week, the new deceiving GitHub PR feature :P Although after the 'needs changes' it needed the 'changes approved'  again on Trac to enable the pqm merge.

needs-review

dumol commented on content/review.rst:
```diff
@@ -54,7 +57,7 @@ General
   names and then do a follow-up branch for the renaming.
 
-* Feel free to submit as many merge request as you can. The size of 
+* Feel free to submit as many merge request as you can. The size of
```
Now that you've touched the line above, I think you should also correct the grammar error in it. :-]

dispera replied:
Hahah, you are right! Corrected :)

dumol commented on content/review.rst:
```diff
@@ -87,12 +90,17 @@ For the person requesting a review
 
+* Before submitting a ticket for review, check that you have documented your
+  work accordingly, for example in the affected repository documentation, or
+  the styleguide.
```
Just a suggestion… Some work is documented in Trac's wiki, especially the work done by the system administrators.

dispera replied:
Alright, added trac's wiki as an example too ;)

dumol commented on content/review.rst:
```diff
@@ -256,7 +264,7 @@ Reviewer's check list - Any Role
 * Is there a release notes entry for changes?
 
-* Is there documentation for changes? 
+* Is there documentation for changes?
```
This sounds bad… How about: _Are the changes documented?_

dispera replied:
Yes, corrected, makes sense. I reordered a bit that section so it is better now too.

brunogola commented:
changes-approved

adiroiban commented:
I don't see @hcs0 approval after her last request for changes 2 days ago... yet this was merged

dispera commented:
Right, after I asked on my previous comment, that was all she wanted to add as she reviewed and approved the other changes and idea before too - and I made the change she requested very quickly. But it appears I should have triggered yet another needs-review round at that point..

> On Dec 31, 2016, at 5:33 AM, Adi Roiban <notifications@github.com> wrote:
> 
> I don't see @hcs0 <https://github.com/hcs0> approval after her last request for changes 2 days ago... yet this was merged

adiroiban commented:
What is done is done... but with that in mind I feel that this PR still don't solve the issue.

I feel that the reason why we do reviews and how we do the review is still not clear... for me it should be clear that once you ask someone for a review, you will wait for a review from that persons... or at least some soft of feedback
